lib/srv/desktop/audit_test.go

```diff
@@ -137,71 +162,25 @@ func TestSessionStartEvent(t *testing.T) {
 		},
 	} {
 		t.Run(test.desc, func(t *testing.T) {
-			s.onSessionStart(
-				context.Background(),
-				emitterPreparer,
-				id,
-				s.cfg.Clock.Now().UTC().Round(time.Millisecond),
-				"Administrator",
-				"sessionID",
-				desktop,
-				test.err,
-			)
-
-			event := emitter.LastEvent()
-			require.NotNil(t, event)
-
-			startEvent, ok := event.(*events.WindowsDesktopSessionStart)
-			require.True(t, ok)
-
+			startEvent := audit.makeSessionStart(test.err)
 			require.Empty(t, cmp.Diff(test.exp(), *startEvent))
 		})
 	}
 }
 
 func TestSessionEndEvent(t *testing.T) {
-	s, id, emitter := setup()
-	emitterPreparer := libevents.WithNoOpPreparer(emitter)
-
-	desktop := &types.WindowsDesktopV3{
-		ResourceHeader: types.ResourceHeader{
-			Metadata: types.Metadata{
-				Name:   "test-desktop",
-				Labels: map[string]string{"env": "production"},
-			},
-		},
-		Spec: types.WindowsDesktopSpecV3{
-			Addr:   "192.168.100.12",
-			Domain: "test.example.com",
-		},
-	}
 
-	c := clockwork.NewFakeClockAt(time.Now())
-	s.cfg.Clock = c
-	startTime := s.cfg.Clock.Now().UTC().Round(time.Millisecond)
-	c.Advance(30 * time.Second)
-
-	s.onSessionEnd(
-		context.Background(),
-		emitterPreparer,
-		id,
-		startTime,
-		true,
-		"Administrator",
-		"sessionID",
-		desktop,
-	)
-
-	event := emitter.LastEvent()
-	require.NotNil(t, event)
-	endEvent, ok := event.(*events.WindowsDesktopSessionEnd)
-	require.True(t, ok)
+	id, audit := setup(testDesktop)
+
+	audit.clock.(clockwork.FakeClock).Advance(30 * time.Second)
+
+	endEvent := audit.makeSessionEnd(true)
```

Comment:
Most of these tests got a bit simpler, as we no longer need to emit the event and then ask a mock emitter what was emitted, we just build the event and it's already the type we expect.